### ISSUE TITLE
Mapping Keyboard Keys to Controller Buttons

### ISSUE THREAD

**Issue by DieProgrammIDE** [xbmc/xbmc#17859]
### Describe the bug
Here is a clear and concise description of what the problem is:

When setting a Mapping for the Keyboard with the Controller (Pressing down on the (General Xbox 360 Controller) should resolve to down on the Keyboard), the mapping doesn't change anything. I need it in game.libretro.dosbox

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Controller down resolves to down key

## Actual Behavior

Nothing happens

### To Reproduce

1. Map a Keyboard Key to a Controller like described in [https://kodi.wiki/view/HOW-TO:Configure_controllers](Kodi Wiki Article)
2. Try it (in DosBox with Giana Sisters :))

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

I also tryed (and resetted) the Mapping XML File from the AddOn and the Controller itself.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [X ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LibreElec 9.2
 - Kodi version: Kodi 18.6 Leia

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DaVukovic** [comment]
Keymaps are either bound to Windows in Kodi or have a "global" setting, which then counts for any window inside Kodi. I doubt hat dosbox has a WindowID in Kodi and therefore I also doubt that a keymapping will work. 

But let's see what @garbear says

**Comment by DieProgrammIDE** [comment]
> Keymaps are either bound to Windows in Kodi or have a "global" setting, which then counts for any window inside Kodi. I doubt hat dosbox has a WindowID in Kodi and therefore I also doubt that a keymapping will work.
> 
> But let's see what @garbear says

OK, but is this mapping with the gui the right way, or do i need to find out the Window ID and Create some File

**Comment by garbear** [comment]
Hi @DieProgrammIDE, unfortunately this isn't implemented yet. Currently, you can only use a keyboard to emulate keyboard keys, and a controller to emulate controllers. If the DOS game supports joysticks, then a controller can be used, otherwise a keyboard is required. Sorry about that, this used to be possible (emulating a keyboard with a controller) but the functionality was removed to simplify the system. It's my goal to re-add this capability, but it might be a while before it's possible.

**Comment by DieProgrammIDE** [comment]
DosBOX does have some Controllers to Emulate, but my game doesn't seem to work with any Controller although i have mapped Buttons on my "Classical PlayStation Layoutet Controller" to the ones supported. Maybe this is a problem with the Game itself (Giana Sisters 32k on an overclocked (2,0Ghz) Raspberry Pi 4).

Is there any way for me to do all of this manually. I've tried editing the Buttonmap (and the other File, listing which controllers are supported, topography i think) but didn't suceed. Is there a Tutorial or any Rule how these Buttonmap Files work and how the styling is supposed to be. Which Options are available etc.

DosBox Zip File: https://mirrors.kodi.tv/addons/leia/game.libretro.dosbox+windows-x86_64/game.libretro.dosbox-0.74.0.5.zip

Although it is not what i would have liked to hear concerning the controller mapping. Still Thank You for answering my issue!

**Comment by garbear** [comment]
Can you try mapping the "2 button joystick" and "4 button joystick"? I believe DOS uses those profiles.

**Comment by DieProgrammIDE** [comment]
That’s what i tried and didn‘t work, as it‘s not implemented like @garbear said, if I understood it the right way.

Are there any updates on this?

> Can you try mapping the "2 button joystick" and "4 button joystick"? I believe DOS uses those profiles.

**Comment by DieProgrammIDE** [comment]
Are there any updates on this?

**Comment by DieProgrammIDE** [comment]
Any updates?

**Comment by garbear** [comment]
No, sorry. Busy with work now, but last summer I made time for RetroPlayer work. I'll try to do the same this summer.

**Comment by garbear** [comment]
I got distracted and built this: https://github.com/xbmc/xbmc/pull/20505

The issue here is still on my list of things to do.

**Comment by garbear** [comment]
Still on my list of things to do, and probably easy fodder for AI.
